### DISCUSSION TITLE
`create-next-app` Not working due to folder permission

### DISCUSSION THREAD

**Discussion by devjiwonchoi** [vercel/next.js#53913] [Help] [answered]
### Error Output

```console
The application path is not writable, please check folder permissions and try again.
It is likely you do not have write permissions for this folder.
```

### Related Next.js Source

https://github.com/vercel/next.js/blob/9229f74b86cfcc2f5d994c67219838fe8394943d/packages/create-next-app/create-app.ts#L127-L135

https://github.com/vercel/next.js/blob/599cde86318d3e92043390f9a90a633a1c8bf376/packages/create-next-app/helpers/is-writeable.ts#L3-L10

### Attempts
- `pnpm create next-app`
- `yarn create next-app`
- `npx create-next-app`
- Reinstalled `homebrew`, `pnpm`
- `chmod -R 777`
- `chown -R`
- Tried `fs.promises.access(process.cwd(), fs.constants.W_OK)`, confirmed returning no error

### Note

It worked fine until I factory reset this Macbook, assuming node related config issue.

### Additional information

```js
- Hardware: Macbook Air M1 2020
- Software: OS X Ventura 13.5
- Node: v18.17.0
- Package Manager: `Homebrew 4.1.4` & `pnpm 8.6.12`
```

**Comment by devjiwonchoi** [comment, marked as answer]
This was caused by naming the project as `@next/project-name`, recommending to add this case on

https://github.com/vercel/next.js/blob/599cde86318d3e92043390f9a90a633a1c8bf376/packages/create-next-app/index.ts#L205-L215

with error `name can only contain URL-friendly characters`, which occurs when naming the project starting with `@`.

### Edit

The dependency `validate-npm-package-name` which Next.js is using passes `@tag/name` format.
There must be an additional escape before reaching the `isWriteable` function.

### Edit2

It was expected behavior, since `create-next-app` allows `project path` in the `What is your project named?` field.
I still do think when asking for the project name, it should warn if the project name includes `/` for clarification.

**Reply by maxpsq** [reply]
I just noticed that also trying to create a new app in the current directory `(.)`no longer works for the very same reason. Just try
`npx create-next-app .`

According to some examples I found in some blogs, in the past it was possible.

**Comment by KHALIL965** [comment]
npx create-next-app created my project but pnmp command its throwing error of not have access of writable

**Comment by anacierdem** [comment]
Similarly `npx create-next-app <path>` doesn't work even for a simple path name without special characters.

**Reply by devjiwonchoi** [reply]
How does it not work? Could you provide steps to repro?

**Reply by snakehugo** [reply]
mkdir testnpm
cd testnpm
npm init -y

package.json will be
{
  "name": "testnpm",
  "private": true,
  "workspaces": ["apps/*", "packages/*"]
}

npm install turbo --save-dev
npx turbo init

npx create-next-app@latest apps/web
default settings

Error

**Reply by snakehugo** [reply]
Okay found it, you just need to create the apps/ folder before running the command, it will not create it on his own :x 
And therefore, it is not really a permission issue but more a non existing folder. 

Just needed a tittle mkdir apps
